### PR TITLE
fix: missing import and definition in callback

### PR DESCRIPTION
**Pull request by anversa-pro**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [ ] :sparkles: Feature
- [x] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

-

## What is the new behavior?

-  add missing import for useCallback fixes the compilation error.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -25,13 +26,14 @@ export default function LoginPage(): JSX.Element {
       const resultToken = await result.user.getIdToken()
       setLoading(false)
       window.localStorage.setItem('token', JSON.stringify(resultToken))
-
+      console.log('----->result', result)
```
Comment (FelipeLosada1996) [inline src/pages/Login/Login.tsx:29]: eliminar console.log

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -58,24 +60,23 @@ export default function LoginPage(): JSX.Element {
...
-  }
-
-  useEffect(() => {
-    handleRedirect()
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  
+    const handleRedirect = useCallback(() => {
+      const { email, isAdmin } = user.state
+       console.log('Redirect', email, isAdmin)
```
Comment (FelipeLosada1996) [inline src/pages/Login/Login.tsx:66]: Todos

File: src/pages/Login/Login.tsx (hunk 1 above)
Comment (anversa-pro) [reply src/pages/Login/Login.tsx]: done

File: src/pages/Login/Login.tsx (hunk 2 above)
Comment (anversa-pro) [reply src/pages/Login/Login.tsx]: done

File: src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { useState, useContext, useEffect } from 'react'
+// import { useState, useContext, useEffect } from 'react'
```
Comment (jlora-mck) [inline src/pages/Login/Login.tsx:1]: Remove comments

File: src/pages/Login/Login.tsx [hunk 4]
```diff
@@ -24,14 +25,14 @@ export default function LoginPage(): JSX.Element {
       const result = await api.login({ email: userName, password })
       const resultToken = await result.user.getIdToken()
       setLoading(false)
-      window.localStorage.setItem('token', JSON.stringify(resultToken))
-
+      window.localStorage.setItem('token', JSON.stringify(resultToken))      
+
```
Comment (jlora-mck) [inline src/pages/Login/Login.tsx:29]: remove space

File: src/pages/Login/Login.tsx [hunk 5]
```diff
@@ -58,24 +59,23 @@ export default function LoginPage(): JSX.Element {
...
+      const { email, isAdmin } = user.state
+      let shouldRedirectTo = '/login'
+  
+      if (email) {
+        shouldRedirectTo = isAdmin ? '/events/list' : '/'
+      } else if (eventId) {
+        shouldRedirectTo = `/event-info/${eventId}`
+      }
+      history.push(shouldRedirectTo)
+    }, [eventId, history, user.state])
```
Comment (jlora-mck) [inline src/pages/Login/Login.tsx:74]: i think we can remove user.state from here

File: src/pages/Login/Login.tsx (hunk 4 above)
Comment (anversa-pro) [reply src/pages/Login/Login.tsx:29]: removed

File: src/pages/Login/Login.tsx (hunk 3 above)
Comment (anversa-pro) [reply src/pages/Login/Login.tsx:1]: removed

File: src/pages/Login/Login.tsx (hunk 5 above)
Comment (anversa-pro) [reply src/pages/Login/Login.tsx:74]: No, is required to trigger the redirect

**Review by jlora-mck — APPROVED** [review]

**Review by jose-sal-y-rosas — APPROVED** [review]
👍🏼

**Review by FelipeLosada1996 — APPROVED** [review]

**Pull request merged**